### PR TITLE
setting up VS Code for a full IDE experience with pretty printing

### PR DESCRIPTION
**Pull request by aaronshim**
Added
 - Updated README instructions for installing VS Code and extensions
 - A directory for VS Code configuration files
 - `launch.json` configuration for VS Code debugging extension targets
 - pretty-printing scripts from the Rust distribution
 - `.gdbinit` to load Rust-GDB pretty-printers plus instructions on how to enable it

### CODE REVIEW

**Comment by aaronshim** [comment]
@rowedonalde, thoughts?

**Comment by rowedonalde** [comment]
Sorry for the wait, @aaronshim. I'll take a look at this over the next day or so.

File: .gdbinit [hunk 1]
```diff
@@ -0,0 +1,13 @@
+add-auto-load-safe-path /home/aaron/Projects/zip-info/.gdbinit
```
Comment (rowedonalde) [inline .gdbinit]: Is there a way to make this general to any user's home folder on any platform? Or can this line just be removed?

**Comment by rowedonalde** [comment]
Sorry this has taken a while. I'm trying this all out with VSCode on my Mac.

File: .gdbinit [hunk 2]
```diff
@@ -0,0 +1,11 @@
+python
```
Comment (rowedonalde) [inline .gdbinit:1]: Which version of Python is it expecting?

**Comment by rowedonalde** [comment]
I'm stuck on getting gdb to pick up on debug symbols (I think) from the binaries, but that's a Mac problem that other people have had that's outside the scope of this PR.

**Pull request merged**